src/Security/Items.cs

```diff
@@ -185,14 +184,13 @@ internal SecKeyChain (NativeHandle handle)
 				else
 					copy.LowlevelSetObject (CFBoolean.TrueHandle, SecItem.ReturnData);
 
-				IntPtr ptr;
-				status = SecItem.SecItemCopyMatching (copy.Handle, out ptr);
+				status = SecItem.SecItemCopyMatching (copy.Handle, out IntPtr ptr);
 				n = null;
 				if (status == SecStatusCode.Success){
 					if (max == 1)
-						return new NSData [] { Runtime.GetNSObject<NSData> (ptr, false)! };
+						return new NSData [] { Runtime.GetNSObject<NSData> (ptr, true)! };
```

Comment:
Not sure what this is supposed to do.

Reply:
To rephrase, it's possible that because we're not looking at this as a container of NSData, we would leak the other elements.

Reply:
The native function is weird: [it only returns an array if the limit is > 1](https://developer.apple.com/documentation/security/1398306-secitemcopymatching?language=objc):

"By default, this function returns only the first match found. To obtain more than one matching item at a time, specify the search key [kSecMatchLimit](https://developer.apple.com/documentation/security/ksecmatchlimit?language=objc) with a value greater than 1. The result will be an object of type [CFArrayRef](https://developer.apple.com/documentation/corefoundation/cfarrayref?language=objc) containing up to that number of matching items."

This means the code is good as-is.